### PR TITLE
MekHQ PortraitChooser

### PR DESCRIPTION
**Pull request by Windchild292**
This moves MekHQ to using the PortraitChooser and starts some of the icon implementation for portraits.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 1]
```diff
@@ -692,20 +685,21 @@ public void setCallsign(String n) {
...
-        return portraitCategory;
+    //region Portrait
+    public AbstractIcon getPortrait() {
+        return portrait;
     }
 
-    public String getPortraitFileName() {
-        return portraitFile;
+    public void setPortrait(AbstractIcon portrait) {
+        this.portrait = portrait;
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/personnel/Person.java:694]: Can this be null? If not lets require it to be non-null here.

**Review by sixlettervariables — APPROVED** [review]
One suggestion, otherwise looks good.

File: MekHQ/src/mekhq/campaign/personnel/Person.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/Person.java:694]: No code using AbstractIcon should be nullable outside of the GUI. That is the base design for AbstractIcon.

Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/Person.java:694]: However, I've added an assert cause here and will open a PR or perhaps PRs to ensure this is the case in MM  too.

**Pull request merged**